morph_tool/axon_point.py

```diff
@@ -0,0 +1,66 @@
+"""Module to detect the terminal point of the main axon/"""
+import logging
+import numpy as np
+
+from neurom import COLS
+from morphio import SectionType, IterType
+
+logger = logging.getLogger("morph_tool")
+
+
+def axon_point_section(morph, direction=None, bbox=None):
+    """Estimate axon point as the terminal point of the main axon.
+
+    This point is defined as the point for which the sum of the angles with the direction vector
+    is minimal. This corresponds to a main axon being a branch that follows best a straight line
+    in the given direction. More constraints can be added with bbox argument, if incorect axon is
+    detected.
+
+    Args:
+        morph (morphio.Morphology): morphology
+        direction (ndarray): estimated direction of main axon, if None [0, -1, 0] is used
+        bbox (dict): bbox of the form {'x': [-100, 100], 'y': [-500, -400]}
+
+    Returns:
+        MorphIO section ID for which the last point is the axon point
+    """
+    if direction is None:
+        direction = [0.0, -1.0, 0.0]
+    else:
+        direction /= np.linalg.norm(direction)
+
+    def _get_angle(section, direction):
```

Comment:
move above to the line 27 before `direction` condition

Reply:
make this function standalone in this module (keep it private)

Reply:
done